tests/tests/tier0/bluechi-is-online-agent-wait/test_bluechi_is_online_agent_wait.py

```diff
@@ -0,0 +1,124 @@
+    bluechi_is_online: BluechiIsOnline,
+    wait_time: int,
+    future: ResultFuture,
+):
+    future.result = bluechi_is_online.agent_is_online(
+        wait_time,
+    )
+
+
+def exec(ctrl: BluechiControllerMachine, nodes: Dict[str, BluechiAgentMachine]):
+
+    node_foo = nodes[NODE_FOO]
+
+    # Test 1: Start agent and verify --wait option returns 0 immediately
+    LOGGER.debug("Starting test number 1 - agent should be online.")
+    with Timeout(
+        IMMEDIATE_RETURN_TIMEOUT, "bluechi-is-online didn't return immediately"
+    ):
+        assert node_foo.bluechi_is_online.agent_is_online(wait=WAIT_PARAM_VALUE)
+
+    # Test 2: Stop agent and verify --wait returns 1 after wait time is expired
+    node_foo.systemctl.stop_unit("bluechi-agent")
+    assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "inactive")
+    LOGGER.debug("Starting test number 2 - agent should remain offline.")
+
+    start_time = time.time()
+    result = node_foo.bluechi_is_online.agent_is_online(wait=WAIT_PARAM_VALUE)
+
+    assert (
+        not result
+    ), f"Expected bluechi-is-online with --wait={WAIT_PARAM_VALUE} to exit false"
+    assert (
+        time.time() - start_time <= WAIT_PARAM_VALUE + 1
+    ), "Expected around 5 second for bluechi-is-online to exit"
+
+    # Test 3: Stop agent, run 'bluechi-is-online agent --wait', start agent and verify --wait returns 0 before the
+    # wait time expires
+    LOGGER.debug(
+        "Starting test number 3, ensure agent is inactive before starting `bluechi-is-online`."
+    )
+    with Timeout(WAIT_PARAM_VALUE, "Timeout during Test 3"):
+        assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "inactive")
+        LOGGER.debug("Agent confirmed inactive after stopping.")
+
+        result_future_wait = ResultFuture()
+        start_time = time.time()
+        LOGGER.debug(
+            "Starting `bluechi-is-online` thread with wait time of 20 seconds."
+        )
+        check_thread_wait = threading.Thread(
+            target=check_agent,
+            args=(
+                node_foo.bluechi_is_online,
+                WAIT_PARAM_VALUE,
+                result_future_wait,
+            ),
+        )
+        check_thread_wait.start()
+        time.sleep(SLEEP_DURATION)
+        node_foo.systemctl.start_unit("bluechi-agent")
+        assert node_foo.wait_for_unit_state_to_be("bluechi-agent", "active")
+        LOGGER.debug("Agent confirmed active after starting.")
+
+        # Wait for the thread to complete
+        with Timeout(2, "Thread didn't stop within 2s of starting bluechi-agent"):
+            check_thread_wait.join()
+
+        elapsed_time = time.time() - start_time
+        LOGGER.debug(
+            f"Test 3 result: {result_future_wait.result}, Elapsed time: {elapsed_time:.2f} seconds"
+        )
+        assert (
+            result_future_wait.result
+        ), "Expected agent to come online before wait expired"
+        assert (
+            2 < elapsed_time < 3
```

Comment:
Shouldn't the check be, that bluechi-is-online as soon as agent is connect, meaning before `WAIT_PARAM_VALUE` timeout passed?

```
        assert (
            elapsed_time < WAIT_PARAM_VALUE
        ), f"bluechi-si-online didn't finish before wait timeout"
```

Reply:
Done.